pkg/binding/bindingpolicy.go

```diff
@@ -67,15 +67,15 @@ const (
 func (c *Controller) handleBindingPolicy(ctx context.Context, obj runtime.Object) error {
 	bindingPolicy, err := runtimeObjectToBindingPolicy(obj)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to convert runtime.Object to BindingPolicy: %s", err)
```

Comment:
%w instead of %s

Reply:
As of Go 1.13 (or earlier if you use [golang.org/x/xerrors](https://godoc.org/golang.org/x/xerrors)), you can use the %w verb which wraps the error such that it can later be unwrapped with [errors.Unwrap](https://golang.org/pkg/errors/#Unwrap), and so that it can be considered with [errors.Is](https://golang.org/pkg/errors/#Is) and [errors.As](https://golang.org/pkg/errors/#As).

Reply:
from chatgpt:

<img width="662" alt="image" src="https://github.com/kubestellar/kubestellar/assets/19717747/f9fbae82-7e9a-4d57-9409-ab90aed00cdf">

Reply:
Yeah, too sleepy. Fixing.